h2quic/client.go

```diff
@@ -169,16 +138,14 @@ func (c *Client) Do(req *http.Request) (*http.Response, error) {
 
 	hasBody := (req.Body != nil)
 
-	c.mutex.Lock()
-	for c.encryptionLevel != protocol.EncryptionForwardSecure {
-		c.cryptoChangedCond.Wait()
-	}
+	<-c.handshakeChan // wait until the handshake has completed
```

Comment:
We should probably have a way to unblock this if an error occurs between dialing and closing the chan.